File: app/src/main/java/com/ktc/sitepulse/domain/WorkerSearch.kt
package com.ktc.sitepulse.domain

import com.ktc.sitepulse.data.model.Worker

/** Mirrors findWorker()/verifyId()'s fuzzy-match behavior from the web app. */
object WorkerSearch {
    fun findExact(workers: List<Worker>, id: String): Worker? =
        workers.find { it.id == id.trim() }

    fun suggest(workers: List<Worker>, query: String, limit: Int = 8): List<Worker> {
        val q = query.trim().lowercase()
        if (q.isEmpty()) return emptyList()
        return workers.filter { it.name.lowercase().contains(q) || it.id.contains(query.trim()) }.take(limit)
    }
}

File: web/js/workerSearch.js
// Mirrors app/src/main/java/com/ktc/sitepulse/domain/WorkerSearch.kt.
export function findExact(workers, id) {
  const trimmed = (id || "").trim();
  return workers.find((w) => w.id === trimmed) || null;
}

export function suggest(workers, query, limit = 8) {
  const q = (query || "").trim().toLowerCase();
  if (!q) return [];
  const raw = (query || "").trim();
  return workers
    .filter((w) => w.name.toLowerCase().includes(q) || w.id.includes(raw))
    .slice(0, limit);
}
